jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/APIKeys/RotateAPIKeysRule.java

```diff
@@ -95,14 +99,20 @@ private boolean rotateApiKeys(String vmEsURL, Map<String, Object> mustFilter) th
         if (hitsJsonArray.size() > 0) {
             logger.info("hit array size {}", hitsJsonArray.size());
             JsonObject apiKeys = (JsonObject) ((JsonObject) hitsJsonArray.get(0)).get(PacmanRuleConstants.SOURCE);
-            String cdDate=apiKeys.get("createdDate").getAsString();
-            SimpleDateFormat sdf=new SimpleDateFormat("EEE MMM dd HH:mm:ss yyyy");
-            TimeZone timeZone=TimeZone.getTimeZone("IST");
-            sdf.setTimeZone(timeZone);
-            cdDate=cdDate.replace("GMT ", "");
-            Date date=  sdf.parse(cdDate);
-           if (TimeUnit.DAYS.convert(new Date().getTime()-date.getTime(), TimeUnit.MILLISECONDS)<=90){
-                    validationResult= true;
+            String createdDateStr = apiKeys.get("createdDate").getAsString();
+            String istZoneIdStr = "Asia/Kolkata";
+            ZoneId istZoneId = ZoneId.of(istZoneIdStr);
+            java.util.Date parsedCreatedDate = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSS")
+                    .parse(createdDateStr);
+            Instant createdDateInstant = parsedCreatedDate.toInstant();
+            LocalDateTime localDateTime = LocalDateTime.ofInstant(createdDateInstant, istZoneId);
+            ZonedDateTime zonedDateTime = localDateTime.atZone(istZoneId);
+            long millis = zonedDateTime.toInstant().toEpochMilli();
+            int ageInDays = (int) TimeUnit.DAYS.convert(new Date().getTime() - millis, TimeUnit.MILLISECONDS);
+            logger.debug("ageInDays:{}", ageInDays);
```

Comment:
it seems like a bit excessive change comparing to what it was before... why date format change brought here so many other changes?

Reply:
Agree it is excessive code changes. Now simplified the approach in latest push